### DISCUSSION TITLE
Apollo DevTools no longer work with two ApolloClient instances

### DISCUSSION THREAD

**Discussion by josoroma-zz** [apollographql/apollo-client-devtools#720] [Ideas]
Apollo DevTools no longer work with two ApolloClient clients, we are using `apollo.tsx` and `apolloRefresh.tsx`, the first one is our global client, which makes use of apolloRefresh.tsx to get a new refresh cookie and token.

![image](https://user-images.githubusercontent.com/69951/133814728-b25846d6-b934-454a-af64-e2b427103bd0.png)

**Comment by hwillson** [comment]
Hmm - if AC devtools worked with multiple clients somehow before, that was fluke. That wasn’t / isn’t a supported feature. The old communication model involved talking to Apollo Client through a global environment variable, and that same environment variable was used by any `ApolloClient` instance running in memory. So if there were 2 AC instances, the most recently instantiated object would have grabbed the global. Maybe it appeared to work if you're instantiating `ApolloClient` in 2 separate parts of your app, but it wouldn’t have been stable as you moved around (e.g. the cache, query/mutation lists, etc. would have kept changing).

We could consider adding this functionality in the future, but that will require a UI overhaul as 2 AC instances means 2 different caches / 2 different query lists / 2 different mutation lists / etc. It would also require us to add some kind of a toggle when using the integrated explorer, so we know which AC instance you want to run your operation with. This could be a large amount of work to accommodate. 

I'll move this to a feature request. Thanks @josoroma!
